waku/node/waku_node.nim

```diff
@@ -401,6 +406,13 @@ proc mountRelay*(node: WakuNode,
   for pubsubTopic in pubsubTopics:
     node.subscribe((kind: PubsubSub, topic: pubsubTopic))
 
+  ## Mount the WakuMetadata protocol
+  ## Notice that we mount the WakuMetadata at this point because there couldn't exist
+  ## WakuRelay without WakuMetadata.
+  node.mountMetadata(metadataClusterId).isOkOr:
```

Comment:
not sure I agree with this. A node can have relay disabled but still mount metadata protocol (eg light client).

My suggestion would be to:
* leave it as it was
* just raise an exception in the peer manager if wakumetadata is nil.

Reply:
Okay perfect!
The thing is that  I need the `WakuMetadata` to be mounted somehow so that I don't have an exception thrown from c-bindings. IMO, that shouldn't throw an exception because of trying to access a `nil` ref object.

I'll re-think the approach then!

Reply:
mm i mean something like this:
https://github.com/waku-org/nwaku/blob/master/waku/node/peer_manager/peer_manager.nim#L429